### ISSUE TITLE
Add support for EBNF

### ISSUE THREAD

**Issue by dwijnand** [github/linguist#3075]

**Comment by arfon** [comment]
:wave: @dwijnand! I'm assuming that `EBNF` is https://en.wikipedia.org/wiki/Extended_Backus–Naur_Form?

If so, we'd welcome a pull request implementing this. You can read more about how to that that here: https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language

**Comment by dwijnand** [comment]
Hey. Thank you for the link. I only mentioned it because I created a gist and couldn't get it to highlight in any way, but I don't use EBNF often enough to look into implementing the highlighting myself.

**Comment by pchaigno** [comment]
@dwijnand I can take care of adding support for BNF if you'd like. I'll just need a few pieces of information:
- Do you have a few BNF files under open source license I could use as samples/examples?
- Is `.bnf` the main file extension? Are there other?
- Is there an official logo or color?

https://github.com/mackoj/language-bnf could be used as a grammar for syntax highlighting.

**Comment by dwijnand** [comment]
* There are sections of EBNF is the Scala spec: https://github.com/scala/scala/search?utf8=%E2%9C%93&q=ebnf
* I would assume `.ebnf`, but I'm not sure
* Doesn't look like there are either.

**Comment by Alhadis** [comment]
Erm, EBNF (and BNF) is less a language than a formalised notation...

**Comment by dwijnand** [comment]
True, but it still has syntax that would be more pleasant to read with some highlighting. Are you saying it's out of scope for linguist?

**Comment by Alhadis** [comment]
Yes, I would say so.

First and foremost, Linguist exists to identify languages (with certain data formats added to stop it misclassifying some languages based on shared file extensions, etc). Managing syntax highlighting is only a secondary responsibility.

**Comment by Crissov** [comment]
Support for EBNF and the `.ebnf` file extension seems to have been added in the meantime, but not BNF `.bnf` or ABNF etc.

**Comment by dwijnand** [comment]
Cool. Thanks for letting me know.

**Comment by ppKrauss** [comment]
Hi @Crissov  and others, `.ebnf` is not working, see [syntax.ebnf file](https://github.com/okfn-brasil/trazdia/blob/master/docs/fragId-syntax.ebnf): no syntax highlighting.

**Comment by Alhadis** [comment]
Well, the file is correctly recognised by Linguist as EBNF:

~~~console
λ trazdia (master): $LING/bin/linguist docs/fragId-syntax.ebnf
fragId-syntax.ebnf: 39 lines (22 sloc)
  type:      Text
  mime type: text/plain
  language:  EBNF
~~~

But this is [happening in Lightshow](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=source.ebnf&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fokfn-brasil%2Ftrazdia%2Fmaster%2Fdocs%2FfragId-syntax.ebnf&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fokfn-brasil%2Ftrazdia%2Fmaster%2Fdocs%2FfragId-syntax.ebnf&code=) too. I suspect there've been changes upstream, in the grammar that's providing EBNF highlights...

I'll investigate.

**Comment by Alhadis** [comment]
Ah wait. You're not looking at an EBNF grammar. :-) I'm not sure what that is, but it isn't EBNF. Notice how the syntax is [very different to the one you're using](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form#Examples). And you're also using DocBlock-style comments, which I've never seen used in an *BNF-type grammar before...

**Comment by ppKrauss** [comment]
Hi @Alhadis, thanks (!). Yes very different, how to obtain syntax highlighting of the popular W3C and http://www.bottlecaps.de/rr/ui syntax?

**Comment by Alhadis** [comment]
[Okay, there's your answer](https://www.w3.org/TR/xquery/#nt-bnf). The EBNF example you showed earlier is actually an "XQuery Grammar", which is an extension of it.

**Comment by ppKrauss** [comment]
Hi @Alhadis, 

Hum... We advance in the consensus, summarizing:

1. you expressed **the fact**: the standard adopted at `github/linguist` for the language name "EBNF" was [ISO&nbsp;14977](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form#Table_of_symbols), so it is not an language, is a dialect, we can say  **the "EBNF-ISO"  dialect**. Other people need (it is  *also popular* and important) other dialect, <br/>the **"EBNF-W3C"** (ref. [TR/xml/#sec-notation](https://www.w3.org/TR/xml/#sec-notation) and [TR/xquery/#nt-bnf](https://www.w3.org/TR/xquery/#nt-bnf))...

2. We, the users of the  EBNF-W3C dialect,  need **the solution**:  syntax highlighting our dialect.

-----

In other words: the semantic of the language label `ebnf` is *"language dialect EBNF-ISO"*, and there are users that need other dialect, the `ebnf-w3c`... How to create the new dialect from the old dialec, with minimal cost of work and time?

**Comment by Alhadis** [comment]
Well, I'd be more than happy to pledge my help,...

**Comment by ppKrauss** [comment]
Thanks @Alhadis (!), I am starting, but only will really work next week... The first task where I need help is to understand how to express space as `\s+`  instead `,`  and how to express `*` and `?` operators over `(...)` gorup, all at [Syntaxes/EBNF.tmLanguage](https://github.com/ppKrauss/EBNF-W3C.tmbundle/blob/master/Syntaxes/EBNF.tmLanguage).

**Comment by ErikBjare** [comment]
Is there anyway to get ebnf highlighting in markdown segments?

i.e. something like

```
\```enbd
char = "c"
\```
```
